ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/invitationServiceAgent.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Inject, Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { NotAuthenticatedServiceAgent } from './notAuthenticatedServiceAgent.service';
+import { ConfigurationService } from '../configuration.service';
+import { LoggingService } from '../logging.service';
+import { EMPTY, Observable, throwError } from 'rxjs';
+import { catchError, map, mergeMap } from 'rxjs/operators';
+import { InvitationCheckPayload } from '../../../../../ddp-testboston/src/app/models/invitationCheckPayload';
+import { DdpError } from '../../../../../ddp-testboston/src/app/models/ddpError';
+import { ErrorType } from '../../../../../ddp-testboston/src/app/models/errorType';
+
+@Injectable()
+export class InvitationServiceAgent extends NotAuthenticatedServiceAgent<any> {
+    private readonly OK_STATUS = 200;
+
+    constructor(
+        @Inject('ddp.config') _configuration: ConfigurationService,
+        _http: HttpClient,
+        _logger: LoggingService) {
+        super(_configuration, _http, _logger);
+    }
+
+    public check(invitationId: string, recaptchaToken: string, zip?: string): Observable<any> {
+        const payload: InvitationCheckPayload = {
+            invitationId,
+            zip,
+            recaptchaToken,
+            auth0ClientId: this.configuration.auth0ClientId};
+        return this.postObservable(`/studies/${this.configuration.studyGuid}/invitation-check`,  payload, {}, true)
+            .pipe(
+                map(response => response ? response.body : null),
+                catchError(err => {
+                    const errorType = err.status === 400 ? ErrorType.InvalidInvitation : ErrorType.Unknown;
+                    return throwError(new DdpError('Problem with submission', errorType));
```

Comment:
Wanted to make sure we treated 400 as validation errors for which we would alert user.